curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md

```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
 
 ## Description
 <section id='description'>
+
 <code>Array.prototype.reduce()</code>, or simply <code>reduce()</code>, is the most general of all array operations in JavaScript. You can solve almost any array processing problem using the <code>reduce</code> method.
-This is not the case with the <code>filter</code> and <code>map</code> methods since they do not allow interaction between two different elements of the array. For example, if you want to compare elements of the array, or add them together, <code>filter</code> or <code>map</code> could not process that.
-The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as a special application of <code>reduce</code>.
-However, before we get there, let's practice using <code>reduce</code> first.
+
+The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as special applications of <code>reduce</code>.
+The <code>reduce</code> method iterates over each item in an array and returns a single value (i.e. string, number, object, array).  This is achieved via a callback function that is called each iteration.
+
+The callback function accepts four arguments.  The first argument is known as the accumulator, which gets assigned the return value of the callback function from the previous iteration.  The second is the current element being processed.  The third is the index of that element.  The fourth is the array upon which <code>reduce</code> was called.
```

Comment:
```suggestion
The callback function accepts four arguments: The first argument is known as the accumulator, which gets assigned the return value of the callback function from the previous iteration, the second is the current element (the element being processed), the third is the index of that element and the fourth is the array upon which <code>reduce</code> is called.
```

Reply:
I am not sure about replace the period with a colon here.  Not sure why we can not just leave it as a period.  It is a simple sentence as it is.

Reply:
Looks better now?

Reply:
I honestly don't see a problem with what was originally there.

Reply:
I felt like the statement you had chosen to describe the second parameter is a bit confusing. See the change I’ve made to that part of the sentence.

Reply:
I feel using `current element` and then in parentheses using the word `the element` is redundant.  I am OK with the commas and changing the `was` to `is` in the last part of the sentence though.

Reply:
Edited.